controllers/audience.go

```diff
@@ -0,0 +1,68 @@
+package controllers
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"reflect"
+
+	"platform-go-challenge/models"
+	Repository "platform-go-challenge/repository"
+	"platform-go-challenge/utils"
+
+	"gorm.io/gorm"
+)
+
+type AudienceController struct{}
+
+func (a AudienceController) UpdateAudience(db *gorm.DB) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		var audience models.Audience
+		var error models.Error
+
+		json.NewDecoder(r.Body).Decode(&audience)
+
+		if reflect.ValueOf(audience.ID).IsZero() || reflect.ValueOf(audience.UserId).IsZero() {
```

Comment:
Is the use of reflection necessary here? Is there any alternative way to perform those validations?

Reply:
Great, that makes sense.